pkg/services/ssosettings/ssosettingsimpl/service.go

```diff
@@ -470,7 +470,9 @@ func mergeSettings(storedSettings, systemSettings map[string]any) map[string]any
 
 	for k, v := range systemSettings {
 		if _, ok := settings[k]; !ok {
-			settings[k] = v
+			if isMergingAllowed(k) {
```

Comment:
Just to be extra defensive, should we also add a check for making sure that the `isMergingAllowed` is only applied to SAML settings?

Reply:
I agree with @dmihai that we for sure should keep this generic and try to not add provider specific things here. Another way do do this would be to extend the provider interface with this functionality, or add an optional interface that providers can implement. That way we can make sure we don't block for other provider in case of key collision.